src/test/java/com/bakdata/common_kafka_streams/util/ErrorCapturingTransformerTopologyTest.java

```diff
@@ -0,0 +1,452 @@
+package com.bakdata.common_kafka_streams.util;
+
+import static org.mockito.Mockito.mock;
+
+import java.util.List;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.Produced;
+import org.apache.kafka.streams.kstream.Transformer;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.jooq.lambda.Seq;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(SoftAssertionsExtension.class)
+class ErrorCapturingTransformerTopologyTest extends ErrorCaptureTopologyTest {
+
+    private static final String ERROR_TOPIC = "errors";
+    private static final String OUTPUT_TOPIC = "output";
+    private static final String INPUT_TOPIC = "input";
+    private static final Serde<String> STRING_SERDE = Serdes.String();
+    private static final Serde<Long> LONG_SERDE = Serdes.Long();
+    private static final Serde<Double> DOUBLE_SERDE = Serdes.Double();
+    private Transformer<Integer, String, KeyValue<Double, Long>> mapper = null;
+
+    @Override
+    protected void buildTopology(final StreamsBuilder builder) {
+        final KStream<Integer, String> input = builder.stream(INPUT_TOPIC, Consumed.with(null, STRING_SERDE));
+        final KStream<Double, ProcessedKeyValue<Integer, String, Long>> mapped =
+                input.transform(() -> ErrorCapturingTransformer.captureErrors(this.mapper));
```

Comment:
As described earlier in the code example, please make `mapper` a supplier and use new transformer objects, whenever the TransformerSupplier is called.

(This comment may apply to all TransformerTests)